TypeSystem/variant.h

```diff
@@ -50,6 +54,16 @@ SOFTWARE.
 
 namespace current {
 
+namespace variant {
+#ifdef VARIANT_CHECKS_AT_RUNTIME_INSTEAD_OF_COMPILE_TIME
+// When variant inner type checks are run-time, no nested variants are allowed.
+using InnerType = CurrentStruct;
```

Comment:
I'd call it this way had it been an in-class type, but it's a namespace-only one.

Think it's OK to just have `current::variant::object_base_t`?